### ISSUE TITLE
Gallery mode jumps to top of thread randomly 

### ISSUE THREAD

**Issue by calactic** [Adamantcheese/Kuroba#1345]
I'm not sure exactly how this is being caused, I've tried to replicate but get inconsistent results, example replication below:

![Screenshot_20210825-100927](https://user-images.githubusercontent.com/78216797/130763054-ae5fb72f-9590-4693-81df-f1d8e99433ed.png)

Last current image of thread visible
![Screenshot_20210825-100934](https://user-images.githubusercontent.com/78216797/130763062-66a5fc62-40c5-44e8-a2aa-78c970d45f92.png)

Gallery appears as intended

![Screenshot_20210825-100945](https://user-images.githubusercontent.com/78216797/130763110-d9fdeda3-e49b-4a86-84ff-35347de242f6.png)

Scroll down so that last thumbnail in thread isn't visible 

![Screenshot_20210825-100954](https://user-images.githubusercontent.com/78216797/130763121-0dd9ce0e-84aa-4382-a231-32e542533cd6.png)

Gallery is back at the start of the thread 
![Screenshot_20210825-101002](https://user-images.githubusercontent.com/78216797/130763142-cd63a49d-ca1d-4812-b1e6-ee4f00ef2d4c.png)

Screenshot of start of the thread.

This happened when no thumbnails were visible on screen at any point in the thread. When the gallery is opened, you can see the gallery scrolling back to the top as it loads every thumbnail. Closing the gallery returns you to the position you opened the gallery from. 

Although I could get this to repeat in the thread I took these screenshots (https://boards.4channel.org/v/thread/568444339) , in other threads it seems to be at random when it will force the gallery to start from the first image. It might be whenever gallery thumbnails are being loaded for the first time.

**Comment by calactic** [comment]
Release c832bc44

**Comment by calactic** [comment]
Gallery works as expected:
a23d519
4e02bc7
3e9528b



Gallery broken:
3f281b5b
21051d78
c832bc44


It looks like any release that has the loading circle when pulling thumbnails has this issue. Not being able to sort dev releases by release date on github is making finding the last properly working release a hassle.

**Comment by ghost** [comment]
Yep I noticed this too

**Comment by samlux04** [comment]
bump. kindly do the needful

**Comment by Adamantcheese** [comment]
So this one's weird, but understandable. Here's the rundown:
Commit fb83c6a is the problem that is causing the strange behavior. The only thing in the commit is an upgrade from Exoplayer 2.14.2 to 2.15.0. The problem therein resides that androidx's recycler view dependency inside of Exoplayer got bumped from 1.1.0 to 1.2.1, which is where the issue is coming from.

It appears that something in [this commit](https://android.googlesource.com/platform/frameworks/support/+/fe867028214a364167c893083065eddcd709e223%5E%21/#F4) caused [this issue](https://issuetracker.google.com/issues/241138586) to appear, which took me a while of scrounging around to find.

tl;dr: Fixed some paddings and it works now. Lots of research to figure it out though. Will push soon™.
